src/main/java/net/snowflake/ingest/utils/Utils.java

```diff
@@ -430,4 +433,31 @@ public static String concatDotPath(String... path) {
     }
     return sb.toString();
   }
+
+  /**
+   * Get the footer size (metadata size) of a parquet file
+   *
+   * @param bytes the serialized parquet file
+   * @return the footer size
+   */
+  public static long getParquetFooterSize(byte[] bytes) throws IOException {
+    final int magicOffset = bytes.length - ParquetFileWriter.MAGIC.length;
+    final int footerSizeOffset = magicOffset - Integer.BYTES;
+
+    if (footerSizeOffset < 0) {
+      throw new IllegalArgumentException(
+          String.format("Invalid parquet file. File too small, file length=%s.", bytes.length));
+    }
+
+    if (!ParquetFileWriter.MAGIC_STR.equals(
+        new String(bytes, magicOffset, ParquetFileWriter.MAGIC.length))) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Invalid parquet file. Bad parquet magic, expected=%s, actual=%s.",
+              ParquetFileWriter.MAGIC_STR,
+              new String(bytes, magicOffset, ParquetFileWriter.MAGIC.length)));
+    }
+
+    return BytesUtils.readIntLittleEndian(bytes, footerSizeOffset);
```

Comment:
is it always little endian irrespective of parquet V1 and parquet V2? How can we add a test that ensures this method is correct?

Reply:
Yes, the V1 and V2 only affects data page. ([ref](https://github.com/apache/parquet-java/blob/42cf31c0fbe4f000d4ddb1e1092c6634989ea3ca/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java#L1880)). For testing, I think we can use `serializedBytes.length - writer.getColumnIndexReference().getOffset() - metadataSize - ParquetFileWriter.MAGIC.length - footerSizeBytes(4) = extenedMetadataSize`. What do you think?

Reply:
Added test for parquet footer size & extendedMetadataSize in `BlobBuilderTest`